docs/api/js/utils.md

```diff
@@ -1,68 +1,785 @@
+
+```ts
+export const CONTRACT_DEPLOYER = new utils.Interface(require('../../abi/ContractDeployer.json').abi);
+```
+
+#### L1 messenger
+
+Used for sending messages from zkSync Era to Ethereum.
+
+```ts
+export const L1_MESSENGER = new utils.Interface(require('../../abi/IL1Messenger.json').abi);
+```
+
+#### L1 and L2 bridges
+
+Bridge interface abis for L1 and L2.
+
+```ts
+export const L1_BRIDGE_ABI = new utils.Interface(require('../../abi/IL1Bridge.json').abi);
+export const L2_BRIDGE_ABI = new utils.Interface(require('../../abi/IL2Bridge.json').abi);
+```
+
+#### L1 to L2 alias offset
+
+Used for applying and undoing aliases on addresses from L1 to L2.
+
+```ts
+export const L1_TO_L2_ALIAS_OFFSET = '0x1111000000000000000000000000000000001111';
+```
+
+#### Magic value
+
+The value returned from `isEIP1271SignatureCorrect` to confirm signature correctness.
+
+```ts
+export const EIP1271_MAGIC_VALUE = '0x1626ba7e';
 ```
 
-zkSync also allow a more friendly zero address alias in our SDK and API:
+#### EIP712 transaction type
+
+Constant representing an EIP712 transaction type.
+
+```ts
+export const EIP712_TX_TYPE = 0x71;
+```
+
+#### Priority op transaction on L2
+
+Constant representing a priority transaction operation on L2.
+
+```ts
+export const PRIORITY_OPERATION_L2_TX_TYPE = 0xff;
+```
+
+#### Max bytecode length
+
+Used for ensuring bytecode length is not over the maximum allowed.
+
+```ts
+export const MAX_BYTECODE_LEN_BYTES = ((1 << 16) - 1) * 32;
+```
+
+### Useful addresses
+
+#### ETH token layer 1
 
 ```typescript
-export const ETH_ADDRESS = "0x0000000000000000000000000000000000000000";
+export const ETH_ADDRESS = '0x0000000000000000000000000000000000000000';
 ```
 
-## ABI of zkSync smart contract
+#### ETH token alias on ZkSync Era
 
 ```typescript
-export const ZKSYNC_MAIN_ABI = new utils.Interface(require("../abi/IZkSync.json"));
+export const L2_ETH_TOKEN_ADDRESS = '0x000000000000000000000000000000000000800a';
 ```
 
-## IERC20 interface
+#### Bootloader
 
-Convenient when interacting with native tokens on zkSync.
+```ts
+export const BOOTLOADER_FORMAL_ADDRESS = '0x0000000000000000000000000000000000008001';
+```
 
-```typescript
-export const IERC20 = new utils.Interface(require("../abi/IERC20.json"));
+#### Contract deployer
+
+```ts
+export const CONTRACT_DEPLOYER_ADDRESS = '0x0000000000000000000000000000000000008006';
+```
+
+#### L1 messenger
+
+```ts
+export const L1_MESSENGER_ADDRESS = '0x0000000000000000000000000000000000008008';
 ```
 
-## Encoding paymaster params
+### Gas
 
-Utility method that returns the correctly formed `paymasterParams` object for the common [paymaster flows](../../dev/developer-guides/aa.md#built-in-paymaster-flows).
+#### `DEFAULT_GAS_PER_PUBDATA_LIMIT`
+
+- Use a large amount of gas for signing on layer 2.
+- The amount ensures any reasonable limit is accepted. 
+
+:::info
+- The operator is NOT required to use the actual value and can use any value up to that signed by the user.
```

Comment:
we want to change this at some point in time in the future

```suggestion
- As of now, the operator is NOT required to use the actual value and can use any value up to that signed by the user. This will be changed in the future.
```

Reply:
That's ok, we tell them then, not now.